### PR TITLE
Fix network simulation and remote peer ID CLI argument parsing

### PR DESCRIPTION
**Pull request by jm-clius**
This PR addresses #342

It contains:
1. Following similar restrictions in `nim-libp2p`, `wakunode2` now applies strict formatting to the `PeerInfo` multiaddresses obtained from parsing CLI arguments. Specifically, `PeerInfo` gets initialised with a `peerId` string and a `multiaddr` wire address. This fixes the `./build/start_network2` network simulation.
2. `./build/chat2` example fixed in a similar manner to (1).
3. Updated `quicksim2` from using the old RPC methods to the new JSON-RPC API.

(1) and (3) have the combined effect of fixing the [quick network simulation](https://github.com/status-im/nim-waku/tree/master/waku/v2#waku-quick-simulation)

### CODE REVIEW

File: examples/v2/chat2.nim [hunk 1]
```diff
@@ -78,8 +78,27 @@ proc initAddress(T: type MultiAddress, str: string): T =
 
 proc parsePeer(address: string): PeerInfo =
```
Comment (staheri14) [inline examples/v2/chat2.nim:79]: Does it have any unit test? if not, might be good to have it tested

**Review by staheri14 — APPROVED** [review]

File: examples/v2/chat2.nim (hunk 1 above)
Comment (jm-clius) [reply examples/v2/chat2.nim:79]: Agree! This is quite a central part of `wakunode2` connectivity. Will add.

Comment (jm-clius) [reply examples/v2/chat2.nim]: Done in 105bbd4.

Also extracted this functionality to a utils module. Cleaning up imports as I go along.

File: tests/v2/test_wakunode.nim [hunk 2]
```diff
@@ -251,3 +254,16 @@ procSuite "WakuNode":
     await node1.stop()
     await node2.stop()
     await node3.stop()
+
+  asyncTest "Peer info parses correctly":
+    ## This is such an important utility function for wakunode2
+    ## that it deserves its own test :)
```
Comment (staheri14) [inline tests/v2/test_wakunode.nim:260]: nice comment! :D

File: waku/v2/node/wakunode2.nim [hunk 3]
```diff
@@ -14,6 +14,7 @@ import
   ../protocol/waku_store/waku_store,
   ../protocol/waku_swap/waku_swap,
   ../protocol/waku_filter/waku_filter,
+  ../utils/peers,
```
Comment (staheri14) [inline waku/v2/node/wakunode2.nim:17]: a minor comment, do we follow any naming conventions to distinguish utils modules from others? e.g., peers_utils or something like this

**Review by staheri14 — APPROVED** [review]
Looks nice!

File: tests/v2/test_wakunode.nim [hunk 4]
```diff
@@ -251,3 +254,16 @@ procSuite "WakuNode":
...
+    
+    let
+      addrStr = "/ip4/127.0.0.1/tcp/60002/p2p/16Uuu2HBmAcHvhLqQKwSSbX6BG5JLWUDRcaLVrehUVqpw7fz1hbYc"
```
Comment (staheri14) [inline tests/v2/test_wakunode.nim:263]: Just came to my mid, you may want to consider some corner cases, with wrong formats (in case such situation may happen), otherwise, it is fine

File: waku/v2/node/wakunode2.nim (hunk 3 above)
Comment (jm-clius) [reply waku/v2/node/wakunode2.nim:17]: As far as I can tell, the (current) convention is simply to group these under a `utils` subfolder. I think there's scope to improve how we handle utils in general, so watch this space. :)

File: tests/v2/test_wakunode.nim (hunk 4 above)
Comment (jm-clius) [reply tests/v2/test_wakunode.nim:264]: Done :) 4b5094d

**Review by oskarth — APPROVED** [review]
